core/src/main/java/com/linecorp/armeria/common/DefaultRpcRequest.java

```diff
@@ -120,11 +120,11 @@ public boolean equals(Object obj) {
             return true;
         }
 
-        if (!(obj instanceof DefaultRpcRequest)) {
+        if (!(obj instanceof RpcRequest)) {
             return false;
         }
 
-        final DefaultRpcRequest that = (DefaultRpcRequest) obj;
+        final RpcRequest that = (RpcRequest) obj;
```

Comment:
I think we should check the exact type here because a user may implement a new subtype of `RpcRequest` which has an additional property. In that case, `equals()` will not be supposed to return `true`.